cumulusci/tasks/bulkdata/delete.py

```diff
@@ -14,6 +14,10 @@ class DeleteData(BaseSalesforceApiTask, BulkJobTaskMixin):
             "description": "A list of objects to delete records from in order of deletion.  If passed via command line, use a comma separated string",
             "required": True,
         },
+        "criteria": {
+            "description": "A criteria using the same syntax as a where-clause",
```

Comment:
It seems atypical that the same criteria can be applied to the queries for all of the sObjects being deleted (if more than one). Do you see the use case as being a single-object run with a WHERE clause, or is there a need to apply WHEREs to multiple sObjects?

Reply:
Sure, I agree that two features of this task are not entirely orthogonal. And I was worried about that at first.

Yes: you're unlikely to use the multi-object and the WHERE features together. 

But trying to make the two features work together nicely very quickly turns into a messy science project. Something like:

objects=account:name LIKE '%foo%',contact:name='foo'

I felt it was better to just let users pick whether they want to take advantage of the multi-object mode or the WHERE mode. If in some obscure case they happen to work together properly for you then that's fine as well.

One can always achieve multi-object deletes like this (using simplified Robot syntax):

Run Task   DeleteData    objects=account   criteria=name LIKE '%foo%
Run Task   DeleteData    objects=contact    criteria=name='foo'

The performance will be almost identical, I think, because the DeleteData task itself is just a loop over the object types.

Reply:
What would you say to just adding a note to the option description to say "This option is only available when deleting a single object's records", and a check when we init options that it's only populated if the length of the object list is 1?

Reply:
I can understand that argument but I do think we should check and throw an exception if you try to specify criteria and multiple objects at the same time.

Also I wonder if we should just call this `where` instead of `criteria`. Criteria makes me think of Salesforce workflow rule criteria which have a different structure. https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/customfield.htm#filteritem

Reply:
Check with exception: 85f710dd54e23f4f376a5f229e50bf28b8732a30

Rename 'criteria' to 'where': b0744bd1aa42c51368cd638a8f07ea49e06f2524